src/main/java/space/obminyashka/items_exchange/controller/UserController.java

```diff
@@ -95,11 +96,19 @@ public List<AdvertisementTitleDto> getCreatedAdvertisements(@Parameter(hidden =
             @ApiResponse(responseCode = "400", description = "BAD REQUEST"),
             @ApiResponse(responseCode = "403", description = "FORBIDDEN")})
     @ResponseStatus(HttpStatus.ACCEPTED)
-    public String updateUserPassword(@Valid @RequestBody UserChangePasswordDto userChangePasswordDto,
-                                     @Parameter(hidden = true) Authentication authentication) throws InvalidDtoException {
-        User user = findUserByValidCredentials(authentication, userChangePasswordDto.getOldPassword());
+    public String updateUserPassword(@Parameter(name = "New password")
+                                     @Size(min = 8, max = 30, message = "{" + INVALID_PASSWORD_SIZE + "}")
+                                     @Pattern(regexp = PatternHandler.PASSWORD, message = "{" + INVALID_PASSWORD + "}")
+                                     @RequestParam String password,
+                                     @Parameter(name = "Confirm new password")
+                                     @RequestParam String confirmPassword,
+                                     @Parameter(hidden = true) Authentication authentication) throws DataConflictException {
+        User user = getUser(authentication.getName());
+        checkPasswordUniqueAndNotUsed(user.getPassword(), password, confirmPassword);
```

Comment:
I think name of this function should be change

Reply:
i like it)